docs/testing_servers.md

```diff
@@ -0,0 +1,340 @@
+trailers to return, any errors to throw, and any data to respond with. All request types contain a response definition 
+which is used to instruct the server how to respond to the request. This response definition can also be unset entirely
+and servers should respond in a fashion specific to their RPC type. The various approaches are outlined below.
+
+Servers must also echo back the received request data in their responses and the method for doing so will vary depending 
+on the type of RPC. The `ConformancePayload` message is a field on all RPC response types. Servers will use this field 
+to echo back these details. This will include:
+
+* Any response data
+* Information observed from the request such as:
+  * Request headers
+  * Any timeout specified
+  * All request bodies received
+  * Any query parameters observed (only applicable for GET operations such as `IdempotentUnary`)
+
+In all, there are six total endpoints you will need to implement. Below is a brief description of each with helpful pseudocode
+for your implementation.
+
+### Unary
+
+The `Unary` endpoint is a unary operation that accepts a single request of type `UnaryRequest` and returns a single 
+response of type `UnaryResponse`. 
+
+The `UnaryRequest` contains a `response_definition` that tells the server how to respond. This definition contains a 
+`oneof` which specifies whether the server should return valid response data or return an error. Servers should also 
+allow this response definition to be unset. In which case, they should set no response headers or trailers and return 
+no response data. The returned conformance payload should only contain the observed request information.
+
+**Pseudocode**
+
+```text
+read the request
+
+capture any request headers and the actual request body as request info
+
+if a response definition was specified then
+  if response definition specifies valid response data then
+
+    build a conformance payload
+
+    set the request info and response data into the conformance payload
+
+  else
+    build an error with the specified error and set the request info into the error details
+
+  set any response headers or trailers indicated in the response definition
+
+else
+  build a conformance payload
+
+  set the request info into the conformance payload
+
+sleep for any specified response delay
+
+return the response with conformance payload or raise the error, depending on which was specified
+```
+
+For the full documentation on implementing the `Unary` endpoint, click [here][unary].
+
+
+### IdempotentUnary
+
+The `IdempotentUnary` endpoint is also a unary operation. It accepts a request of type  `IdempotentUnaryRequest` and returns
+a single response of type `IdempotentUnaryResponse`. It should be handled in mostly the same way as `Unary`.
+
+The only major difference is that this endpoint should be invoked via an HTTP `GET`. As a result, there is no request body
```

Comment:
I think it's worth clarifying this a little more. In particular that implementations that only support gRPC or gRPC-Web do not need to implement it. Also, Connect implementations that do not support GET requests also don't need to implement it.